### PR TITLE
8294187: RISC-V: Unify all relocations for the backend into AbstractAssembler::relocate()

### PR DESCRIPTION
**Pull request by zhengxiaolinX**
Unify all relocations into AbstractAssembler::relocate() as a simple cleanup.

Tested hotspot tier1 and tier2 on QEMU.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8294187](https://bugs.openjdk.org/browse/JDK-8294187): RISC-V: Unify all relocations for the backend into AbstractAssembler::relocate()

### Reviewers
 * [Feilong Jiang](https://openjdk.org/census#fjiang) (@feilongjiang - Author) ⚠️ Review applies to [21ba6a61](https://git.openjdk.org/jdk/pull/10384/files/21ba6a611b8de528fa4a625b8086ac89eca62e81)
 * [Yadong Wang](https://openjdk.org/census#yadongwang) (@yadongw - Author) ⚠️ Review applies to [21ba6a61](https://git.openjdk.org/jdk/pull/10384/files/21ba6a611b8de528fa4a625b8086ac89eca62e81)
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10384/head:pull/10384` \
`$ git checkout pull/10384`

Update a local copy of the PR: \
`$ git checkout pull/10384` \
`$ git pull https://git.openjdk.org/jdk pull/10384/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10384`

View PR using the GUI difftool: \
`$ git pr show -t 10384`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10384.diff">https://git.openjdk.org/jdk/pull/10384.diff</a>

</details>

### CODE REVIEW

**Review by feilongjiang — APPROVED** [review]
LGTM

**Review by yadongw — APPROVED** [review]
Looks fine. But it's best to point out that it's bridging to the compressed-instruction-related relocation.

File: src/hotspot/cpu/riscv/c1_LIRAssembler_riscv.cpp [hunk 1]
```diff
@@ -2145,7 +2145,7 @@ void LIR_Assembler::typecheck_lir_store(LIR_OpTypeCheck* op, bool should_profile
 }
 
 void LIR_Assembler::add_debug_info_for_branch(address adr, CodeEmitInfo* info) {
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/c1_LIRAssembler_riscv.cpp:2147]: I see this function is never called: void LIR_Assembler::add_debug_info_for_branch(address adr, CodeEmitInfo* info)
So I would suggest remove this function. Then following two functions introduced by this PR would become unnecessary:
  void relocate(address at, RelocationHolder const& rspec, int format = 0) {
    code_section()->relocate(at, rspec, format);
  }
  void relocate(address at, relocInfo::relocType rtype, int format = 0, jint method_index = 0) {
    code_section()->relocate(at, rtype, format, method_index);
  }

**Review by RealFYang — CHANGES_REQUESTED** [review]

**Comment by zhengxiaolinX** [comment]
> Looks fine. But it's best to point out that it's bridging to the compressed-instruction-related relocation.

Thanks, I've removed the relocate() definitions and will add them to other patches so this patch could be a simple cleanup. The title and description get updated.

File: src/hotspot/cpu/riscv/c1_LIRAssembler_riscv.cpp (hunk 1 above)
Comment (zhengxiaolinX) [reply src/hotspot/cpu/riscv/c1_LIRAssembler_riscv.cpp]: Thanks for pointing this out. All done.

**Review by RealFYang — APPROVED** [review]
Updated changes look good. Thanks.

**Comment by zhengxiaolinX** [comment]
Thank you all for the review!

/integrate

**Pull request closed without merging**